### ISSUE TITLE
Using FluentValidation instead of FluentValidation.AspNetCore

### ISSUE THREAD

**Issue by lonix1** [micro-elements/MicroElements.Swashbuckle.FluentValidation#97]
Hello @petriashev, thanks for this great library! I've been using it for a while.

I am now using FluentValidation inside a [Mediatr](https://github.com/jbogard/MediatR) pipeline. So now I reference `FluentValidation` instead of `FluentValidation.AspNetCore` (otherwise I will get double validation, once by MVC and once inside Mediatr).

Of course that causes this warning:
> WRN | MicroElements.Swashbuckle.FluentValidation.FluentValidationRules | ValidatorFactory is not provided. Please register FluentValidation.

And of course the validation details are not shown in Swagger.

However I am still using FluentValdation, just in a different way. My validator classes are still the same.

Can you think of a way to get this to work?

**Comment by petriashev** [comment]
Can you create simple repro with Mediatr?

Main class `FluentValidationRules` uses DI for its dependencies. Try to register `IValidatorFactory` for example `HttpContextServiceProviderValidatorFactory` or standard `ServiceProviderValidatorFactory`

**Comment by lonix1** [comment]
Thanks!!! **Please tell me if I understand correctly?**

- when using `FluentValidation.AspNetCore` library, it [automatically registers](https://github.com/FluentValidation/FluentValidation/blob/0e45f4efbab956d84f425b0b7d207ada516720bd/src/FluentValidation.AspNetCore/FluentValidationMvcExtensions.cs#L78) `ServiceProviderValidatorFactory` as implementation of `IValidatorFactory`
- but when using the basic `FluentValidation.DependencyInjectionExtensions` library, that is [not registered](https://github.com/FluentValidation/FluentValidation/blob/0e45f4efbab956d84f425b0b7d207ada516720bd/src/FluentValidation.DependencyInjectionExtensions/ServiceCollectionExtensions.cs#L51)
- so we must manually register it like so:
  ````cs
  services.TryAddTransient<IValidatorFactory, ServiceProviderValidatorFactory>();
  services.AddFluentValidationRulesToSwagger();
  ````

---
If correct, it would be a good idea to add it to [docs](https://github.com/micro-elements/MicroElements.Swashbuckle.FluentValidation#common-problems-and-workarounds) ("Common problems and workarounds"). Example:

**Problem: I'm using `FluentValidation.DependencyInjectionExtensions` instead of `FluentValidation.AspNetCore`**
If you are using the more basic FluentValidation library, then it will not automatically register `IValidatorFactory` and you will get error at runtime, like "ValidatorFactory is not provided. Please register FluentValidation." In that case you must register it manually:
````cs
services.TryAddTransient<IValidatorFactory, ServiceProviderValidatorFactory>();
services.AddFluentValidationRulesToSwagger();
````

See issue 97 for more details.

**Comment by CollinAlpert** [comment]
When using this workaround, I am getting the following warning: `CS0618: Interface 'FluentValidation.IValidatorFactory' is obsolete`. Is this a known issue?

**Comment by petriashev** [comment]
> IValidatorFactory

Yes, `IValidatorFactory` is obsolete. But it's not needed in latest version of package

**Comment by CollinAlpert** [comment]
What version is the "latest"? I am using Microelements.Swashbuckle.FluentValidation version 5.7.0 with FluentValidation.AspNetCore version 11.2.2 and get the message "ValidatorFactory is not provided. Please register FluentValidation." if I don't call `services.TryAddTransient<IValidatorFactory, ServiceProviderValidatorFactory>();`.

**Comment by petriashev** [comment]
@CollinAlpert Try 6.0.0-beta.1

**Comment by CollinAlpert** [comment]
Thanks so much, that fixed it!
